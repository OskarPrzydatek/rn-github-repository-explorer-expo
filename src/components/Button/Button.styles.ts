import { colors } from '@/constants';
import { StyleSheet } from 'react-native';

export const buttonStyles = StyleSheet.create({
  button: {
    padding: 12,
    borderRadius: 6,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.green,
  },
  label: {
    fontWeight: 'bold',
    color: colors.white,
  },
});
