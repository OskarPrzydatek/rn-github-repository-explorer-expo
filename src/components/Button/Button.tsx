import React from 'react';

import {
  ActivityIndicator,
  Platform,
  Text,
  TouchableHighlight,
} from 'react-native';

import { buttonStyles } from '@/components/Button/Button.styles';
import { colors } from '@/constants';

interface IButton {
  label: string;
  isLoading: boolean;
  onPress: () => void;

  testID?: string;
  activityIndicatorTestID?: string;
}

export const Button = ({
  label,
  isLoading,
  onPress,

  testID,
  activityIndicatorTestID,
}: IButton) => {
  const isIOSActivityIndicatorStyles =
    Platform.OS === 'ios' && buttonStyles.activityIndicator;

  return (
    <TouchableHighlight
      style={buttonStyles.button}
      activeOpacity={0.7}
      underlayColor={colors.activeGreen}
      onPress={onPress}
      testID={testID}
    >
      {isLoading ? (
        <ActivityIndicator
          color={colors.white}
          size={19.5} // Android only
          style={isIOSActivityIndicatorStyles}
          testID={activityIndicatorTestID}
        />
      ) : (
        <Text style={buttonStyles.label}>{label}</Text>
      )}
    </TouchableHighlight>
  );
};
