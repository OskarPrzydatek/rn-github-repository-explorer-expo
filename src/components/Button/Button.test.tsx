import { render, screen } from '@testing-library/react-native';

import { Button } from '@/components/Button/Button';

describe('Button', () => {
  it('should match snapshot', () => {
    const tree = render(
      <Button label="Label" isLoading={false} onPress={jest.fn()} />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render activity indicator when loading', () => {
    render(
      <Button
        label="Label"
        isLoading={true}
        onPress={jest.fn()}
        activityIndicatorTestID="button-activity-indicator"
      />,
    );

    const activityIndicator = screen.getByTestId('button-activity-indicator');

    expect(activityIndicator).not.toBeNull();
  });
});
