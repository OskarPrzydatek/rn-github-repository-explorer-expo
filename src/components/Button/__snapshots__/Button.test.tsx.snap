// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1e883e",
      "borderRadius": 6,
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontWeight": "bold",
      }
    }
  >
    Label
  </Text>
</View>
`;
